log-levels.md

```diff
@@ -0,0 +1,46 @@
+# Log Levels 
+
+This guide serves as guideline for library authors with regards to what [SwiftLog](https://github.com/apple/swift-log) log levels are apropriate for use in libraries, and in what siuations to use what level.
+
+## Guidelines for Libraries
+
+SwiftLog defines the following 7 log levels: `trace`, `debug`, `info`, `notice`, `warning`, `error`, `critical`.
+
+Out of those, generally only levels _lower than_ info (exclusive) are generally okey to be used by libraries.
+In the followig section we'll explore how to use them in practice.
+
+### Recommended log levels
+
+It is always fine for a library to log at `trace` and `debug` levels, and these two should be the primary levels any library is logging at.
+
+As `trace` is the finest log level, and generally is not going to be used by end-users of a library unless debugging a very specific issue, it should be a way for developers of the library to "log everything we could possibly need to diagnose a hard to reproduce bug."
+
+**// TODO: should we suggest that a system should be able to survive with debug logging turned on in production (ktoso: I'd suggest that, yes).**
+
+This is in contrast with `debug` which is some users _may_ choose to run with enabled on their production systems. As such, debug logging should not be overly noisy, and provide a high value understanding of what is going on in the library for end users, without diving deep into its internals (which `trace` is intended for).
+
+### Log levels to avoid
+
+All these rules are only _general_ guidelines, and as such may have exceptions to the rules. Please consider the following examples and rationale for why logging at high log levels by a library may not be desirable:
```

Comment:
Nit - tighten up the language
```suggestion
All these rules are only _general_ guidelines, and as such may have exceptions.Consider the following examples and rationale for why logging at high log levels by a library may not be desirable:
```

Reply:
We might want to even remove this paragraph as it's duplicated below

Reply:
Ok will do :)